### PR TITLE
Add support of base64://CONTENT in -tidb-config, -tikv-config and -pd-config

### PR DESCRIPTION
**Pull request by ofey404**
### What problem does this PR solve? 

https://github.com/pingcap/tipocket/issues/324

### What is changed and how does it work?

`-tikv-config`, `-tidb-config`, `-pd-config` support base64 encoded configuration.

Example:
```bash
bin/bank2 ...{other flags}... -tikv-config="base64://bG9nLWxldmVsID0gIndhcm5pbmciCg=="
```

Generate base64 string
```bash
$ cat tikv-test-config.toml ; base64 tikv-test-config.toml 
log-level = "warning"
bG9nLWxldmVsID0gIndhcm5pbmciCg==
```

### Check List 

Tests 

 - Unit test

Code changes

 - Has Go code change

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
add base64 support of -tikv-config, -tidb-config, -pd-config
eg:
    -tikv-config="base64://bG9nLWxldmVsID0gIndhcm5pbmciCg=="
```

### CODE REVIEW

File: pkg/test-infra/tidb/operation.go [hunk 1]
```diff
@@ -459,6 +468,37 @@ func getTidbDiscoveryService(tc *v1alpha1.TidbCluster) *corev1.Service {
 	}
 }
 
+func parseConfig(config string) (string, error) {
```
Comment (cosven) [inline pkg/test-infra/tidb/operation.go:471]: How about checking scheme first and then fallback to old behaviour?
I think we need not to check if the config file exists here. It is checked in `readFileAsString`.
Think about the case when the config value is empty.

File: pkg/test-infra/tidb/operation.go [hunk 2]
```diff
@@ -459,6 +468,37 @@ func getTidbDiscoveryService(tc *v1alpha1.TidbCluster) *corev1.Service {
...
+		}
+		var configData string = ""
+		switch u.Scheme {
+		case "base64":
+			// When parse url like "base64://BASE64CONTENT", BASE64CONTENT would be parsed in url.Host
+			configData, err = readBase64AsString(u.Host)
+			if err != nil {
+				return "", err
+			}
+			fmt.Print(configData)
```
Comment (cosven) [inline pkg/test-infra/tidb/operation.go:495]: remove the debug statement

**Review by cosven — COMMENTED** [review]
rest LGTM

File: pkg/test-infra/tidb/operation.go (hunk 1 above)
Comment (ofey404) [reply pkg/test-infra/tidb/operation.go:471]: You are right.

File: pkg/test-infra/tidb/operation.go (hunk 2 above)
Comment (ofey404) [reply pkg/test-infra/tidb/operation.go]: removed.

File: pkg/test-infra/tidb/operation.go [hunk 3]
```diff
@@ -459,6 +467,32 @@ func getTidbDiscoveryService(tc *v1alpha1.TidbCluster) *corev1.Service {
...
+	}
+	switch u.Scheme {
+	case "base64":
+		// An workaround: When parse url like "base64://BASE64CONTENT", BASE64CONTENT would be parsed in url.Host
+		configData, err = readBase64AsString(u.Host)
+		if err != nil {
+			return "", err
+		}
+	default:
+		return "", fmt.Errorf("not a supported scheme: %s\nsupported schemes like base64://BASE64CONTENT", config)
```
Comment (cosven) [inline pkg/test-infra/tidb/operation.go:490]: Generally, we do not insert `newline` in error message.

**Review by cosven — APPROVED** [review]

File: pkg/test-infra/tidb/operation.go (hunk 1 above)
Comment (mahjonp) [reply pkg/test-infra/tidb/operation.go:470]: I agree with you.

**Review by mahjonp — APPROVED** [review]
LGTM 💯

File: pkg/test-infra/tidb/operation.go (hunk 3 above)
Comment (ofey404) [reply pkg/test-infra/tidb/operation.go]: Have removed it.

**Pull request merged**